app/main/helpers/frameworks.py

```diff
@@ -425,3 +428,46 @@ def validator(self, *args, **kwargs):
                                   "company details.")
 
         return True
+
+
+def return_404_if_applications_closed(data_api_client_callable):
+    def real_decorator(func):
+        @wraps(func)
+        def decorated_view(*args, **kwargs):
+            data_api_client = data_api_client_callable()
+            if 'framework_slug' not in kwargs:
+                current_app.logger.error("Required parameter `framework_slug` is undefined for the calling view.")
+                abort(500, "There was a problem accessing this page of your application. Please try again later.")
```

Comment:
Are we going to start writing custom error messages for 500s now? If so, has this gone past content?

Reply:
That's a good point. I might just drop the customness. Seems unnecessary/doesn't add anything.